### ISSUE TITLE
Allow continuation of lines

### ISSUE THREAD

**Issue by drewbo** [mapbox/mapbox-gl-draw#605]
After a line is added, it can only be edited by moving the vertices or adding midpoints. It would be a nice feature to be able to continue adding vertices like `draw_line_string` but from one of the endpoints. Proposed API would be either:
- `Draw.changeMode('draw_line_string, { id: [id of feature to continue from] })`
- `Draw.changeMode('draw_line_string, { feature: [feature to continue from or add and continue from] })`

Open questions:
- How to specify which endpoint to continue from
- How to incorporate this in the UI (if at all)

**Comment by mcwhittemore** [comment]
@drewbo I'm pumped about this feature!

`coord_path` is exposed on all Positions in a feature. This is a string of `.` separated array indexes and is used in the `direct_select` mode to let users drag one of these Positions.

I'd propose this API:

`Draw.changeMode('draw_line_string, { feature: 'id', from: 'coord_path' })`.

We might want to enforce `coord_path` to either be the `first` or `last` position.

We should make sure this works with `MultiLineStrings`.

**Comment by drewbo** [comment]
@mcwhittemore I like the idea of enforcing `first`/`last` or `start`/`end` for the position. Thinking about it more, users will likely not know which end of the line they are looking to extend from without doing `.get(id)` so does it make sense to have them send a coordinate instead?

**Comment by mcwhittemore** [comment]
`first`/`last` and `start`/`end` only works for LineStrings and while this PR is focused on LineStrings, we need to make sure this can work for Polygons later. `coord_path` solves this.

This isn't a hard thing to compute, but it is a hard thing to explain. I've been trying to think about how util function would make this easier but don't think it can. Either it would return some new list of coordinates for a feature which is overly verbose and still requires you to know how to get the `coord_paths` out of the list (think MultiPolygon vs LineString) or you will need to pass in references to the until function to define what you want back which is really the same thing as creating the coor_path yourself.

TL;DR: lets use coord_paths and figure out how to document them well? Maybe a drawing or something? I feel like this is something a picture would capture better than words.

**Comment by drewbo** [comment]
How will Polygon "continuation" be different than the current Polygon editing? Does it detach a vertex and allow drawing from there? I'm not sure there is as convincing a use case for Polygons to deviate from their current editing mode.

**Comment by mcwhittemore** [comment]
You make a great point... :man_facepalming: 

I was thinking about adding inner rings which won't require a `coord_path`.

**Comment by mcwhittemore** [comment]
MultiLineString still has this problem. Currently `draw_line_string` only handles `LineStrings` though.

**Comment by mcwhittemore** [comment]
closed via 001a23e7544e60a3e06406c7488b1e384aed0801
